Solutions/VMWareESXi/Analytic Rules/ESXiRootLogin.yaml

```diff
@@ -17,14 +17,23 @@ tactics:
 relevantTechniques:
   - T1078
 query: |
+  let p_lookback = 14d;
+  let t_lookback = 1h;
+  let root_ips = VMwareESXi
+  | where TimeGenerated between (ago(p_lookback) .. ago(t_lookback))
+  | where SyslogMessage has_all ('UserLoginSessionEvent', 'root', 'logged in')
+  | extend SrcIpAddr = extract(@'root@(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})', 1, SyslogMessage)
+  | summarize makeset(SrcIpAddr);
   VMwareESXi
+  | where TimeGenerated > ago(t_lookback)
   | where SyslogMessage has_all ('UserLoginSessionEvent', 'root', 'logged in')
   | extend SrcIpAddr = extract(@'root@(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})', 1, SyslogMessage)
+  | where SrcIpAddr in (root_ips)

```

Comment:
Wondering, since we are doing login from uncommon IP address should we not be doing '!in' instead of 'in' here